src/Mod/Sketcher/Gui/CommandConstraints.cpp

```diff
@@ -1671,16 +1672,22 @@ class DrawSketchHandlerDimension : public DrawSketchHandler
 
     void finalizeCommand()
     {
+        if (hasBeenAborted()) {
+            resetTool();
+            return;
+        }
+
         // Ask for the value of datum constraints
         ParameterGrp::handle hGrp = App::GetApplication().GetParameterGroupByPath("User parameter:BaseApp/Preferences/Mod/Sketcher");
         bool show = hGrp->GetBool("ShowDialogOnDistanceConstraint", true);
         const std::vector<Sketcher::Constraint*>& ConStr = Obj->Constraints.getValues();
 
         bool commandHandledInEditDatum = false;
-        for (int i = numberOfConstraintsCreated - 1; i >= 0; i--) {
-            if (show && ConStr[ConStr.size() - 1 - i]->isDimensional() && ConStr[ConStr.size() - 1 - i]->isDriving) {
+        for (int i = cstrIndexes.size() - 1; i >= 0; --i) {
+            int index = cstrIndexes[i];
```

Comment:
Consider using either `#include <boost/range/adaptor/reversed.hpp>` in a range-based for loop, or the iterator-based version using `rbegin()` and `rend()`.